ROX-16283: Reorganize Cloud install docs (#59525)
openshift/openshift-docs: cloud_service/getting-started-rhacs-cloud-ocp.adoc

:_content-type: ASSEMBLY
[id="getting-started-rhacs-cloud-ocp"]
= Getting started with RHACS Cloud Service
include::modules/common-attributes.adoc[]
:context: getting-started-rhacs-cloud-ocp

toc::[]

[role="_abstract"]

> {product-title-managed} ({product-title-managed-short}) provides security services for your {osp} and Kubernetes clusters. See xref:../installing/acs-installation-platforms.adoc#acs-installation-platforms[Supported platforms and installation methods] for more information on supported platforms for secured clusters.

Reviewer: ```suggestion
{rh-rhacscs-first} provides security services for your {osp} and Kubernetes clusters. See xref:../installing/acs-installation-platforms.adoc#acs-installation-platforms[Supported platforms and installation methods] for more information on supported platforms for secured clusters.
```

.Prerequisites

* Ensure that you can access the *Advanced Cluster Security* menu option from the {cloud-console}.
+
[NOTE]
====

> To access the {product-title-managed-short} console, you need your Red Hat Single Sign-On (RH-SSO) credentials, or credentials for another identity provider if that has been configured. See xref:../cloud_service/getting-started-rhacs-cloud-ocp.adoc#default-access-acs-console[Default access to the ACS console].

Reviewer: ```suggestion
To access the {product-title-managed-short} console, you need your Red Hat Single Sign-On (SSO) credentials, or credentials for another identity provider if that has been configured. See xref:../cloud_service/getting-started-rhacs-cloud-ocp.adoc#default-access-acs-console[Default access to the ACS console].
```

. On each {osp} cluster, xref:../cloud_service/installing_cloud_ocp/init-bundle-cloud-ocp-apply#init-bundle-cloud-ocp-apply[apply the init bundle] by using it to create resources.
. On each {osp} cluster, xref:../cloud_service/installing_cloud_ocp/cloud-install-operator#cloud-install-operator[install the {product-title-short} Operator].
. On each {osp} cluster, xref:../cloud_service/installing_cloud_ocp/install-secured-cluster-cloud-ocp.adoc#installing-sc-operator-cloud-ocp[install secured cluster resources in the `stackrox` project] by using the Operator.
. xref:../cloud_service/installing_cloud_ocp/verify-installation-cloud-ocp.adoc#verify-installation-cloud-ocp[Verify installation] by ensuring that your secured clusters can communicate with the ACS instance.

To secure {osp} clusters by using Helm charts or the `roxctl` CLI, perform the following steps:

. . Verify that the clusters you want to secure meet the xref:../cloud_service/installing_cloud_ocp/prerequisites-cloud-ocp.adoc#prerequisites-cloud-ocp[prerequisites].
. In the {cloud-console}, xref:../cloud_service/installing_cloud_ocp/cloud-create-instance-ocp.adoc#cloud-create-instance-ocp[create an *ACS Instance*].

> . On each {osp} cluster you want to secure, xref:../cloud_service/installing_cloud_ocp/cloud-ocp-create-project.adoc#cloud-ocp-create-project[create a project named `stackrox`]. This project will contain the resources for {product-title-managed-short} secured clusters.

Author: I know we generally like to use present tense, but I think since the project is created at this point, but doesn't contain any files yet, it's important to have the context (from "will") that it will store the files that are installed later.

[NOTE]
====
To learn how authentication providers work in ACS, visit xref:../operating/manage-user-access/understanding-authentication-providers.adoc[Understanding authentication providers].
====

A dedicated OIDC client of `sso.redhat.com` is created for each ACS Console. All OIDC clients share the same `sso.redhat.com` realm.
Claims from the token issued by `sso.redhat.com` are mapped to an ACS-issued token as follows:

* `realm_access.roles` to `groups`
* `org_id` to `rh_org_id`
* `is_org_admin` to `rh_is_org_admin`
* `sub` to `userid`

The built-in Red Hat SSO authentication provider has the required attribute `rh_org_id` set to the organization ID assigned to account of the user who created the {product-title-managed-short} instance.
This is the ID of the organizational account the user is a part of. This can be thought of as the "tenant" the user is under and owned by.
Only users with the same organizational account can access the ACS console by using the Red Hat SSO authentication provider.

[NOTE]
====
To gain more control over access to your ACS Console, configure another identity provider instead of relying on the Red Hat SSO authentication provider.

> You can set up your own identity provider after provisioning by using one of the guides under *Operating -> Managing user access*.

Reviewer: ```suggestion
You can set up your own identity provider after provisioning. For more information, see link:"Understanding authentication providers".
```

====
[NOTE]
====

> To configure this authentication provider to be the first authentication option on the login page, its name should be lexicographically smaller than `Red Hat SSO`.

Reviewer: ```suggestion
To configure this authentication provider to be the first authentication option on the login page, its name should be lexicographically smaller than the `Red Hat SSO`.
```

Author: I'm going to check with Eng but leave this as is for now.

====

The minimum access role is set to `None`. Assigning a different value to this field gives access to the {product-title-managed-short} instance to all users with the same organizational account.

Other rules that are set up in the built-in Red Hat SSO authentication provider include the following:

* Rule mapping your `userid` to `Admin`
* Rules mapping administrators of the organization to `Admin`

> You can add more rules to grant access to the ACS Console to someone else with the same organizational account; for example, by using `email` as a key.

Author: This looks new but it's really pre-existing content - good catch, I'll fix!

= Getting started with RHACS Cloud Service
include::modules/common-attributes.adoc[]
:context: getting-started-rhacs-cloud-ocp

toc::[]

[role="_abstract"]

{product-title-managed} ({product-title-managed-short}) provides security services for your {cloud-family} and Kubernetes clusters. This topic provides instructions for using {product-title-managed-short} to secure clusters that are running on supported platforms. See xref:../installing/acs-installation-platforms.adoc#acs-installation-platforms[Supported platforms and installation methods] for more information on supported platforms for secured clusters.

:FeatureName: {product-title-managed-short}
include::snippets/field-trial.adoc[leveloffset=+1]

.Prerequisites

* Ensure that you can access the *Advanced Cluster Security* menu option from the {cloud-console}.
+
[NOTE]
====
To access the {product-title-managed-short} console, you need your Red Hat Single Sign-On (RH-SSO) credentials, or credentials for another identity provider if that has been configured. See xref:../cloud_service/getting-started-rhacs-cloud-ocp.adoc#default-access-acs-console[Default access to the ACS console].
====

[id="installation-overview-acs-cloud"]

> == High-level overview of installation steps

Reviewer: ```suggestion
== Overview of installation steps
```

Author: This language matches the headings in other sections (for example, /installing/installing_ocp/install-rhacs-ocp.adoc) - and it is a high-level overview, as opposed to detailed (because the details are provided in the links), so I am going to leave as is.

Reviewer: :+1:

> == High-level overview of installation steps

Author: same as above

> == High-level overview of installation steps

Reviewer: :+1:
